django/thunderstore/repository/models/package.py

```diff
@@ -257,6 +262,18 @@ def post_save(sender, instance, created, **kwargs):
     def post_delete(sender, instance, **kwargs):
         invalidate_cache_on_commit_async(CacheBustCondition.any_package_updated)
 
+    @staticmethod
+    def ensure_namespace(sender, instance, **kwargs):
+        from thunderstore.repository.models.namespace import Namespace
+
+        try:
+            instance.namespace
+        except Package.namespace.RelatedObjectDoesNotExist:
+            instance.namespace, created = Namespace.objects.get_or_create(
+                name=instance.owner.name, team=instance.owner
+            )
+
 
+signals.pre_save.connect(Package.ensure_namespace, sender=Package)
```

Comment:
Let's not use signals for this (or in general for _anything_ other than features that are allowed to fail or have no other way of handling).

Signals make the code base much harder to follow because you can't clearly see what causes what to happen.

In this case, we don't want to automatically populate a namespace when a package is created, but instead we need to make sure whatever is creating packages populates that information appropriately. The test suite we have should be a good guide to showing what fails if you remove the signal, giving clear indication to what should be changed.

In this case, I don't think there are any other places in code creating packages except the package submission form. The form is used for all the APIs as well as the old frontend if I'm not mistaken.